Packs/VaronisSaaS/Integrations/VaronisSaaS/VaronisSaaS.py

```diff
@@ -0,0 +1,2108 @@
+                criteria_match = False
+                if criteria[key] and len(criteria[key]) > 0:
+                    for value in criteria[key]:
+                        if isinstance(value, str) and value in str(item[key]) or value == item[key]:
+                            criteria_match = True
+                            break
+                    if not criteria_match:
+                        isMatch = False
+                        break
+            if isMatch:
+                filtered_items.append(item)
+
+        return filtered_items
+
+    filtered_items = filter_threat_model_items(mapped_items, {
+        'ID': id_int,
+        'Name': name,
+        'Category': category,
+        'Severity': severity,
+        'Source': source
+    })
+
+    outputs = dict()
+    outputs['ThreatModel'] = filtered_items
+
+    readable_output = tableToMarkdown('Varonis Threat Models', filtered_items, headers=[
+                                      'ID', 'Name', 'Category', 'Severity', 'Source'])
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='Varonis',
+        outputs_key_field='ID',
+        outputs=outputs
+    )
+
+
+def fetch_incidents_command(client: Client, last_run: Dict[str, datetime], first_fetch_time: Optional[datetime],
+                            alert_status: Optional[str], threat_model: Optional[str], severity: Optional[str]
+                            ) -> Tuple[Dict[str, Optional[datetime]], List[dict]]:
+    """This function retrieves new alerts every interval (default is 1 minute).
+
+    :type client: ``Client``
+    :param client: Http client
+
+    :type last_run: ``Dict[str, datetime]``
+    :param last_run:
+        A dict with a key containing the latest alert ingest time we got from last fetch
+
+    :type first_fetch_time: ``Optional[datetime]``
+    :param first_fetch_time:
+        If last_run is None (first time we are fetching), it contains
+        the datetime on when to start fetching incidents
+
+    :type alert_status: ``Optional[str]``
+    :param alert_status: status of the alert to search for.
+        Options are 'New', 'Under investigation', 'Action Required', 'Auto-Resolved' or 'Closed'
+
+    :type threat_model: ``Optional[str]``
+    :param threat_model: Comma-separated list of threat model names of alerts to fetch
+
+    :type severity: ``Optional[str]``
+    :param severity: severity of the alert to search for. Options are 'High', 'Medium' or 'Low'
+
+    :return:
+        A tuple containing two elements:
+            next_run (``Dict[str, Optional[int]]``): Contains last fetched id.
+            incidents (``List[dict]``): List of incidents that will be created in XSOAR
+    :rtype: ``Tuple[Dict[str, int], List[dict]]``
+
+    """
+
+    threat_model_names = argToList(threat_model)
+
+    incidents: List[Dict[str, Any]] = []
+
+    if first_fetch_time is None:
+        raise ValueError("first_fetch_time can't be None")
```

Comment:
what if this is the first time ever used, and lastrun is None?

Reply:
This parameter must be set on setup screen of the instance.
Default value is 1 week so when it's a first run it will request incidents for 1 week back.